packages/core/src/html/UlNodeProcessor.ts

```diff
@@ -0,0 +1,352 @@
+      ],
+      next: undefined,
+      prev: undefined,
+      parent,
+    };
+  } else if (isImage) {
+    const altText = icon.split('/').pop()?.split('.')[0] || '';
+    const divNode: NodeOrText = {
+      type: 'tag',
+      name: 'div',
+      attribs: {
+        style: 'padding: 0.2rem;',
+      },
+      children: [],
+      next: undefined,
+      prev: undefined,
+      parent,
+    };
+
+    const imgNode: NodeOrText = {
+      type: 'tag',
+      name: 'img',
+      attribs: {
+        ...className && { class: className },
+        src: icon,
+        alt: altText,
+        style: defaultSize,
+      },
+      children: [],
+      next: undefined,
+      prev: undefined,
+      parent: divNode,
+    };
+    divNode.children?.push(imgNode);
+    child = divNode;
+  } else if (isOcticon) {
+    const octiconIcon = getOcticonIcon(iconName!);
+    let svgStr = octiconIcon.toSVG({
+      class: octiconClass ? `${iconClass} ${octiconClass} ${className}` : `${iconClass} ${className}`,
+    });
+
+    if (octiconColor) {
+      svgStr = svgStr.replace('<path', `<path fill="${octiconColor}"`);
+    }
+    child = {
+      type: 'tag',
+      name: 'span',
+      attribs: {
+        'aria-hidden': 'true',
+        style: `${defaultSize}margin-right:5px;${size.fontSize ? `font-size:${size.fontSize};` : ''}`,
+      },
+      children: [
+        {
+          type: 'text',
+          data: svgStr,
+        },
+      ],
+      next: undefined,
+      prev: undefined,
+      parent,
+    };
+  } else {
+    child = {
+      type: 'tag',
+      name: 'span',
+      attribs: {
+        class: className ? `${icon} ${className}` : icon,
+        'aria-hidden': 'true',
+        style: `${defaultSize}margin-right:5px;${size.fontSize ? `font-size:${size.fontSize};` : ''}`,
+      },
+      children: iconName ? [
+        {
+          type: 'text',
+          data: iconName,
+        },
+      ] : [],
+      next: undefined,
+      prev: undefined,
+      parent,
+    };
+  }
+  return child;
+}
+
+function getSize(widthPar: string, heightPar: string) {
+  const width = widthPar || heightPar || '25px';
+  const height = heightPar || widthPar || '25px';
+  const sizeUnitRegexp = /(\d+)(px|em|rem|%|vw|vh|vmin|vmax|ex|ch)/;
+  const widthMatch = width.match(sizeUnitRegexp);
+  const heightMatch = height.match(sizeUnitRegexp);
+
+  if (!widthMatch || !heightMatch) {
+    throw new Error(`Invalid width or height format. 
+                    Expected format is number followed by unit
+                    (px | em | rem |%| vw | vh | vmin | vmax | ex | ch).`);
+  }
+
+  // Ensure both width and height are using the same unit
+  if (widthMatch[2] !== heightMatch[2]) {
+    throw new Error('The units of width and height should be the same.');
+  }
+
+  let fontSize = Math.min(parseFloat(widthMatch[1]), parseFloat(heightMatch[1]));
+
+  // Assuming the single character font will take up approximately 70% of the container's smaller dimension
+  fontSize *= 0.7;
```

Comment:
is this to make the size more "standard" amongst the different types of icons? (the ones using font size and the ones not)

Reply:
This only impacts icons utilizing font size, hence images unaffected.